include/onnxruntime/core/framework/data_types.h

```diff
@@ -208,18 +209,25 @@ class DataTypeImpl {
   static MLDataType GetDataType(const std::string&);
 
   static const std::vector<MLDataType>& AllTensorTypes();
+  static const std::vector<MLDataType>& AllTensorTypesWithFloat8();
```

Comment:
nit: it's a bit odd to have `AllX` and then `AllXWithFloat8`, and have the former be a subset. we're using a looser interpretation of "all".

another option is to replace them with `AllXExceptFloat8` and `AllX`, respectively.

similar comment for onnxruntime/core/framework/element_type_lists.h

Reply:
In onnx, I introduced something like AllTensorTypesIR4 and AllTensorTypeIR9 for all type up to IR version 4 or 9 assuming the list of types would be growing. I did not change the definition of AllTensorTypes because the support for float 8 is limited to a short list of operators. I wanted to minimize the impact on the C API of onnxruntime and avoid changing the current meaning of these functions and add new ones instead. That way, all kernels using this type set are not impacted and in most cases, they don't support float 8 types (like all element wises kernel, gemm, ...).

Reply:
I think renaming them similarly here would be good too, but it would touch a lot of code. until they get renamed, could we add comments elaborating on what "All" means here? e.g., all types before float8 types were introduced

Reply:
I kept the existing functions to avoid breaking the API but I introduced new ones suffixed with IR4 and IR9 as I did in onnx package. The existing functions are calling the IR4 versions. We can deprecate them in a future version.